dot/network/transaction.go

```diff
@@ -119,36 +120,54 @@ func decodeTransactionHandshake(_ []byte) (Handshake, error) {
 	return &transactionHandshake{}, nil
 }
 
-func (s *Service) createBatchMessageHandler(txnBatch chan *BatchMessage) NotificationsMessageBatchHandler {
-	return func(peer peer.ID, msg NotificationsMessage) (msgs []*BatchMessage, err error) {
+func (s *Service) createBatchMessageHandler(txnBatchCh chan *BatchMessage) NotificationsMessageBatchHandler {
+	go func() {
+		protocolID := s.host.protocolID + transactionsID
+		ticker := time.NewTicker(1 * time.Second)
+
+		for {
+		out:
+			select {
+			case <-s.ctx.Done():
+				return
+			case <-ticker.C:
+				innerTicker := time.NewTicker(300 * time.Millisecond)
```

Comment:
where did this time come from? and what's the purpose of it exactly?

Reply:
The external tick triggers at a regular interval of `slotDuration`. The inner tick was to trigger timeout, so that it doesn't take the entire `slotDuration`. I have updated the code to use a timer.